components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/ReadinessHandlerTest.kt

```diff
@@ -0,0 +1,50 @@
+/*
+  Copyright (C) 2013-2023 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers
+
+import com.hotels.styx.api.Eventual
+import com.hotels.styx.api.HttpRequest
+import com.hotels.styx.api.HttpResponse
+import com.hotels.styx.api.HttpResponseStatus.OK
+import com.hotels.styx.api.HttpResponseStatus.SERVICE_UNAVAILABLE
+import com.hotels.styx.requestContext
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+import reactor.core.publisher.toMono
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.text.Charsets.UTF_8
+
+class ReadinessHandlerTest : StringSpec() {
+    init {
+        "indicates readiness" {
```

Comment:
this test is testing 2 different scenarios. Please split them into 2 separate tests.

Reply:
It is testing the scenario that the ReadinessHandler changes its content when the readiness changes, but I can also test the two states individually. I'll add those tests.

Reply:
yeah you are right